pypeit/core/arc.py

```diff
@@ -231,34 +231,39 @@ def detect_lines(censpec, nfitpix=5, nonlinear=None, debug=False):
         plt.plot(xrng, detns_smth, 'r-', drawstyle='steps')
         plt.show()
 
-
+    #detns = detns_smth
     # Find all significant detections
     # TODO -- Need to add nonlinear back in here
     pixt = np.where((detns > 0.0) &  # (detns < slf._nonlinear[det-1]) &
                     (detns > np.roll(detns, 1)) & (detns >= np.roll(detns, -1)) &
                     (np.roll(detns, 1) > np.roll(detns, 2)) & (np.roll(detns, -1) > np.roll(detns, -2)) &#)[0]
-                    (np.roll(detns, 2) > np.roll(detns, 3)) & (np.roll(detns, -2) > np.roll(detns, -3)))[0]
-#                    (np.roll(detns_smth, 3) > np.roll(detns_smth, 4)) & (np.roll(detns_smth, -3) > np.roll(detns_smth, -4)) & # )[0]
+                    (np.roll(detns, 2) > np.roll(detns, 3)) & (np.roll(detns, -2) > np.roll(detns, -3))&#)[0]
+                    (np.roll(detns, 3) > np.roll(detns, 4)) & (np.roll(detns, -3) > np.roll(detns, -4)))[0]# & # )[0]
 #                    (np.roll(detns, 4) > np.roll(detns, 5)) & (np.roll(detns, -4) > np.roll(detns, -5)))[0]
-    tampl, tcent, twid = fit_arcspec(xrng, detns, pixt, nfitpix)
+    tampl, tcent, twid, centerr = fit_arcspec(xrng, detns, pixt, nfitpix)
     ww = np.where((~np.isnan(twid)) & (twid > 0.0) & (twid < 10.0/2.35) & (tcent > 0.0) & (tcent < xrng[-1]))
     if debug:
         # Check the results
         plt.clf()
         plt.plot(xrng, detns, 'k-')
         plt.plot(tcent, tampl, 'ro')
         plt.show()
-    return tampl, tcent, twid, ww, detns
+    if return_errors:
```

Comment:
I might suggest you always return errors and make them
`None` if you don't care about them.

I prefer the number of items returned be the same when possible.

Reply:
And I suspec the doc string needs updating either way.

Reply:
Done!